api/handlers/processors.py

```diff
@@ -0,0 +1,89 @@
+import database.database
+from database.models import Processor
+from fastapi import APIRouter, Depends, HTTPException
+from pymongo.collection import Collection
+
+router = APIRouter()
+
+
+@router.post("/api/processors", status_code=201, response_model=Processor)
+def create_processor(
+    processor: Processor,
+    processors: Collection[Processor] = Depends(
+        database.database.processors_collection
+    ),
+):
+    """
+    Create a new processor. The processor will be henceforth identified by name which
+    must be unique or this function will return a 400.
+    """
+    if processors.find_one({"name": processor.name}):
+        raise HTTPException(status_code=409, detail="Processor Already Exists")
```

Comment:
What happens when this exception is raised? Are there options to retry?

Reply:
This is expected to be raised there is even a test for that :)